### ISSUE TITLE
PropertyTypeCoercion false positive when using early exit

### ISSUE THREAD

**Issue by mauriciofauth** [vimeo/psalm#6200]
```php
<?php
final class FooBar
{
    /** @psalm-var positive-int */
    private int $foo = 1;
    /** @psalm-var positive-int */
    private int $bar = 1;
    public function setFoo(int $foo): void
    {
        if ($foo <= 0) {
            return;
        }
        $this->foo = $foo;
    }
    public function setBar(int $bar): void
    {
        if ($bar > 0) {
            $this->bar = $bar;
        }
    }
}
```
```
Psalm output (using commit 89ee1f1): 

ERROR: PropertyTypeCoercion - 13:22 - $this->foo expects 'positive-int',  parent type 'int' provided
```

https://psalm.dev/r/9125d9de31

**Comment by weirdan** [comment]
Duplicate of #6020
